Add example sentences and JMnedict

mymro opened the pull request:
Add example sentences using the Tatoeba project and build JMnedict dictionary for kindle

mymro committed:
Add example sentences

Add example sentences using the Tatoeba project

mymro committed:
Build JMnedict for kindle

mymro committed:
rename mobi files in makefile

mymro commented:
I opted to created two separate dictionaries for JMdict and JMnedict, as it makes searching for terms easier.

mymro committed:
Add combined dictionary and turn cover creation into a function

mymro committed:
Make make file useable on windows and add option for all sentences

mymro committed:
Update readme, bugfix in jmdict.py, add arguments and dialect indicators

mymro committed:
minify html and fix bug where entries are not displayed completely

mymro committed:
add htmlmini to travis.yml; ad the other dictionaries too

mymro committed:
Fix in makefile

mymro committed:
Fix typing error in travis.yml

mymro committed:
Update README and reduce size of html files to fit more sentences

mymro committed:
Merge branch 'Development'

jrfonseca commented:
@mymro thanks for doing this!  I'll review and get this merged.   I'll try to get it done this weekend.  There's a lot in this PR.

One preliminary remark: please note I'm selling this ebook on Kindle store -- https://www.amazon.com/dp/B01G6TK0QW -- not for the money (I'm selling it for the smallest fee Amazon allows me and I plan to donate the very little proceeds I get to JMDdict/EDICT project), but just to make it easier for people to find this ebook, as Amazon doesn't allow to put free ebooks on its store (at least not for indeterminate period), and I even link to this project so people can download it without paying anything.

Please confirm you're OK with eventually update the Kindle store book with your contributions too.

Going forward I'll make a mention of this on README so future contributors are aware.

mymro commented:
@jrfonseca No hurry, I already have the updated dictionaries ;-). As for selling it on the Amazon store: I was aware of that and since it is so cheap and the whole thing is also available for free on GitHub, I am totally OK with you selling it. I think it is actually great to have this dictionary also on the Kindle store,  since a lot of people might not be aware of GitHub.

mymro committed:
Fix find sentence search strategy

mymro committed:
Reduce max sentences in combined dictionary to 0

Having fixed adding sentences they are now more, meaning the dictionary is again too big

mymro committed:
Merge branch 'Development'

mymro commented:
Sorry for the additional commits, but I made some mistakes when searching for the sentences

lippmann commented:
Much obliged for the brilliant work from both of you guys! But where can I download the latest version of the dictionary? I just checked out @mymro 's [updated version](https://github.com/mymro/jmdict-kindle) of the project, but the latest files on the download page are still the ones released in 2018. 

Am I missing something or it's just that the latest one hasn't been uploaded yet? I'm no expert on tech stuff so I'm here to ask about it.

mymro commented:
@lippmann My pull request has not been merged yet by @jrfonseca. As soon as he merges it there will probably be a new release. 

For now, I have just recompiled them with the new code. You can download the [here](https://drive.google.com/drive/folders/1DZ99VpEWkpScbeFzVsnvLm4ZscZgQ6e2?usp=sharing). Please be aware that the combined dictionary does not contain example sentences. Adding those would push the mobi beyond the file size limit.

lippmann commented:
@mymro Thank you very much! The dictionaries work perfectly on my Kindle.

jrfonseca commented:
I was planning to tweak things a bit, but I don't have time, so I've just glanced it, and merged as is for now. Can always tweak later.

Thanks for this!

I'll make a release later so mobi will be easily available.

BTW, it looks I won't be able to update Kindle store anymore -- https://groups.yahoo.com/neo/groups/edict-jmdict/conversations/messages/5859

jrfonseca commented on dictionary.py:
```diff
@@ -88,22 +104,23 @@ def remove(self, reading):
 def write_index_header(stream):
     stream.write('<?xml version="1.0" encoding="UTF-8"?>\n')
     stream.write('<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">\n')
-    stream.write('<html xmlns:idx="www.mobipocket.com" xmlns:mbp="www.mobipocket.com" xmlns="http://www.w3.org/1999/xhtml">\n')
```
I just noticed you forgot to close the html element!   No idea how this even worked since.

Anyway, we can't pretend this is XML any longer, as XML requires quotes on attributes.
